internal/scripting/plugin.go

```diff
@@ -0,0 +1,70 @@
+package script
+
+import (
+	"fmt"
+
+	"github.com/kelindar/talaria/internal/encoding/typeof"
+)
+
+// ------------------------------------------------------------------------------------------------------------
+type computedFuncType func(map[string]interface{}) interface{}
+
+// Go represents a go computed column
+type GoFunc struct {
+	name       string      // Name of the column
+	typ        typeof.Type // The type of the column
+	pluginFunc computedFuncType
+}
+
+// newGoFunc ..
+func newGoFunc(name string, function computedFuncType) *GoFunc {
+	return &GoFunc{
+		name:       name,
+		pluginFunc: function,
+	}
+}
+
+// Name returns the name of the column
+func (c *GoFunc) Name() string {
+	return c.name
+}
+
+// Type returns the type of the column
+func (c *GoFunc) Type() typeof.Type {
+	return typeof.String
+}
+
+// Value computes the column value for the row
+func (c *GoFunc) Value(row map[string]interface{}) (interface{}, error) {
+	fmt.Println("!!!!!!!!!!!GO GOGOGOGOGOGOGOGOGOGOGOGOGOGOGOGOGOGO")
+	// return computeRow(row), nil
+	out := c.pluginFunc(row)
+	return out, nil
+
+	// switch c.typ {
```

Comment:
We still need to make sure we can support both `lua` and `go plugin` ways

Reply:
@kelindar This `Value` is a function of the receiver `GoFunc`, so I think it doesn't matter to `lua` and the comment code about type assertion seems not useful to me.